### ISSUE TITLE
TypeError: undefined is not an object (evaluating 'choice.label')

### ISSUE THREAD

**Issue by usersina** [hossein-zare/react-native-dropdown-picker#47]
Getting the error "TypeError: undefined is not an object (evaluating 'choice.label')" when using the DropDownPicker example.

**To Reproduce**
Using react hooks:

`const [state, setState] = useState({country: 'test'});`
`<DropDownPicker
        items={[
          {label: 'UK', value: 'uk'},
          {label: 'France', value: 'france'},
        ]}
        defaultValue={state.country}
        containerStyle={{height: 40}}
        style={{backgroundColor: '#fafafa'}}
        dropDownStyle={{backgroundColor: '#fafafa'}}
        onChangeItem={(item) =>
          setState({
            country: item.value,
          })
        }
      />`

**Comment by hossein-zare** [comment]
Hello,

Your initial value for country is `test` , It has to be `uk` or `france`.
```javascript
const [state, setState] = useState({country: 'uk'});
```

**Comment by DhineshKSD** [comment]
@hossein-zare @UserSina 
I am facing the same issue [ TypeError: undefined is not an object (evaluating 'choice.label') ]. 

case:
I set the initial state to " ".

How to set the initial state if my dropdown items are dynamically populated from an API.

**Comment by usersina** [comment]
**TL;DR Version:**

Don't set the initial state to `" "` or `null`, instead:
```JAVASCRIPT
useState([{label: 'default_value', value: 'my_default'}])
```
And in the `DropDownPicker`:
```JAVASCRIPT
<DropDownPicker
    ...
    defaultValue="my_default"
/>
```


**The long version (with a loader):**

If you don't want the default value to be displayed in the UI, you can include a state that renders the UI with the dropdown only if the data finish loading, something like this:
```JAVASCRIPT
// The items that are in the dropdown
const [items, setItems] = useState([{label: 'default_value', value: 'my_default'}])

// Initially loading, data is not yet fetched
const [isLoading, setIsLoading] = useState(true);

useEffect(() => {
  getMyData().then(
    (data) => {
      // Success: data fetched successfully
      let processedData = processMyData(data);
      setItems(processedData);
      setIsLoading(false);
    },
    (error) =>{
      // Error: could not fetch data for whatever reason
    }
  )
},[])

// The dropdown view
  const pickerView = () => {
    if (isLoading) {
      return (
        <View>
          <ActivityIndicator animating={isLoading} style={styles.loading} />
        </View>
      );
    } else {
      return (
        <DropDownPicker
            ...
            items={items}
            defaultValue="my_default"
        />
      );
    }
  };

// Finally your render
return(
  <View>
    {pickerView()}
  </View>
)
```

**Comment by DhineshKSD** [comment]
@UserSina 
Thanks for the response.

**Comment by karimxpert** [comment]
@UserSina you missed the processMyData function.

**Comment by ptaberg** [comment]
The mistake appears when you pass defaultValue which doesn't exist in items prop. I think we should make a developer-friendly error message which will not make him surf the web to figure out what's going on.
